### ISSUE TITLE
FOV second eye, objects wrongly not rendered on edge of screen?

### ISSUE THREAD

**Issue by friggasaint** [googlevr/cardboard#54]
I'm observing objects wrongly not being rendered on the second camera, when objects are about to leave the FOV. 
Does anybody else also have this problem?
![Screenshot_20200603-164010 2](https://user-images.githubusercontent.com/1319045/84023698-defd2580-a988-11ea-9d09-df21ad50debd.png)

**Comment by RossMck** [comment]
I have same issue,

**Comment by jballoffet** [comment]
I’m not able to reproduce this issue. Could you please provide a link to a Unity project where this issue is shown in order for me to test it? Please also provide:
- The required steps to expose the issue.
- The version of Unity being used.
- The phone being used.
- The viewer’s parameters being used.

**Comment by JacobBWaggoner** [comment]
I am seeing the same issue in Unity 2019.3.12 on a Pixel 2 without changing the viewer perameters. It happens in any scene if objects move too far to one side of the view.

**Comment by friggasaint** [comment]
I'm already getting this on a fresh project only conducting the quickstart steps for unity
unity 2019.3.13f
Sony Xperia XZ1 Compact
not changing viewer parameters

**Comment by friggasaint** [comment]
My impression has become that the effect grows stronger with smaller screens.
We tested this on an old Iphone SE with a 4 inch screen, which seemingly gave an even worse effect than on my Sony Xperia XZ1 Compact (4.7 inch screen).
On a HTC U Ultra (5.7inch) we couldn't see any issues.
We tested all without changing the viewer params.

One problem might be that one camera is squinted too strongly inside.

**Comment by sammyvrmadman** [comment]
Can confirm that I have the same issue on unity 2019.4.1.1f with an Iphone SE.

**Comment by jballoffet** [comment]
Thanks for the details. We could not reproduce the bug in a Pixel 3XL. We would appreciate the confirmation of Cardboard V1 Viewer (default viewer) to know the device parameters in use. We will prioritise this issue along the others.

**Comment by dletozeun** [comment]
Stumbling upon this issue, this makes me think of a problem with Occlusion Culling. Could you try disabling it on the camera? If you want to keep it you might need to make it again but maybe you can't use it properly in stereo.

**Comment by sammyvrmadman** [comment]
> Thanks for the details. We could not reproduce the bug in a Pixel 3XL. We would appreciate the confirmation of Cardboard V1 Viewer (default viewer) to know the device parameters in use. We will prioritise this issue along the others.

When no other QR code is scanned does that mean the default is selected ? if so I can confirm that this happens on the Cardboard V1 Viewer (default viewer)

**Comment by BrianPickens** [comment]
I've run into the same problem. Here is some info on it:

Repro Steps
-Download the linked GitHub Project
-Build the iOS project
-Use xCode 12.1 to build onto phone
-Open App
-Deny Camera Permissions (still occurs if you allow permission and scan a cardboard viewer)
-Look around and watch the objects on the left side of the right camera. They will disappear before they leave the players view.

Unity 2020.1.11f1
iPhone 7

[GitHub Project Link]( https://github.com/BrianPickens/VR-Bug.git)
[iPhone 7 Screen Record](https://github.com/googlevr/cardboard/files/5515662/VRBugVideo.zip)

**Comment by whizzkid** [comment]
Same problem  here.
from the looks of it, it seems to me that if an object is culled for the left eye after leaving the screen, it's also culled for the right. which still has the object in the camera's frustum and shouldnt cull
unity 2019.4.14f1,  iphone SE

**Comment by jballoffet** [comment]
Sorry for the long delay here. Thanks @BrianPickens for the repro project, I confirm that I’m able to reproduce the issue on an iPhone XR. It has been added to our backlog and will be prioritized along with other issues. We'll update this issue once we have some updates about it.

**Comment by jeff-rey-unity** [comment]
Same on iPhone 6s

**Comment by RuudvanReenen** [comment]
I can confirm the same issue here with an iPhone 6S and Cardboard SDK 1.4.1, Unity 2019.4.18f1. 
Left camera always renders correctly, but the right camera doesn't render objects that aren't visible in the left camera anymore, (at the left edge of the right camera view).

Is there a fix or workaround for this?

**Comment by RuudvanReenen** [comment]
Noticed also that the right camera can see more of what's on the left than the left camera. You can see this clearly in the image of the first post. Are the cameras reversed?

Can this issue be related to https://github.com/googlevr/cardboard/issues/207 where the math goes wrong?

**Comment by Layfully** [comment]
Can someone have a look at this the issue is still there. Is there any hack to prevent it from happening? Is this happening because of frustum culling?

**Comment by Vincze01** [comment]
Same problem here.
Unity 2020.3.15f2, iPhone SE

**Comment by drfede** [comment]
I found a solution, not the best one but it worked for me, I found out the issue it's caused by the camera's frustum which gets distorted, so what I did is manually modify the culling matrix of the camera to COMPLETELY disable the frustum culling.
Briefly what I did is:

```  
      void Start()
        {           
             UnityEngine.Rendering.RenderPipelineManager.beginCameraRendering += OnBeginCameraRendering;
        }

       private void OnBeginCameraRendering(ScriptableRenderContext context, Camera camera)
        {
            camera.cullingMatrix = Matrix4x4.Ortho(-99999, 99999, -99999, 99999, 0.001f, 99999) *
                    Matrix4x4.Translate(Vector3.forward * -99999 / 2f) *
                    camera.worldToCameraMatrix;
        }
        
  ```

**Comment by Layfully** [comment]
This is good start! I will try to look into this deeper. My solution was to increase mesh bounds of every mesh. Your solution is simpler, but less performant.

**Comment by jballoffet** [comment]
A fix for this issue has been submitted and it will be included in v1.12.0. Thanks for reaching out!

**Comment by xinyunh0929** [comment]
This was fixed in v1.12.0.
